frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts

```diff
@@ -103,13 +111,6 @@ export const capabilitiesToUISchema = (capabilities: SpecCapability[] = []) => {
     return k8sResourceCapabilityToUISchema(k8sResourceCapability);
   }
 
-  const fieldDependencyCapability = _.find(capabilities, (capability) =>
-    capability.startsWith(SpecCapability.fieldDependency),
-  );
-  if (fieldDependencyCapability) {
-    return fieldDependencyCapabilityToUISchema(fieldDependencyCapability);
-  }
-
```

Comment:
This was another potential bug. The field dependency should be applied independent of widget or field. The way this was written, a k8sResourcePrefix descriptor would take precedence over fieldDependency, and fieldDependency would take precedence over all other field and widget types. So you would never be able to have custom widget with a dependency.